projects/batfish/src/main/java/org/batfish/bdp/AbstractRib.java

```diff
@@ -430,18 +440,7 @@ final boolean containsRoute(R route) {
 
   @Override
   public final Set<R> getRoutes() {
-    if (_finalRoutes != null) {
-      return _finalRoutes;
-    }
-    return _tree.getRoutes();
-  }
-
-  /**
-   * Freeze the RIB. Prevents addition (merging) of new routes. Also computes and caches the set of
-   * all routes for quick subsequent access.
-   */
-  public void freeze() {
-    _finalRoutes = Collections.unmodifiableSet(getRoutes());
+    return _allRoutes;
```

Comment:
What is the expected semantics of the return value here? 

Questions:

1. Should clients be able to modify the return value?
2. Should clients see updated values if this AbstractRib is modified later?

Reply:
1. They should not.
2. Good question. I believe the answer here is also no since we use getRoutes to iterate over a particular snapshot of the RIB.